controllers/edgedevice_controller.go

```diff
@@ -69,33 +71,48 @@ func (r *EdgeDeviceReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 	logger.Info("Reconciling", "edgeDevice", edgeDevice)
 
-	// create object bucket claim for edge-device
-	if edgeDevice.Status.DataOBC == nil || len(*edgeDevice.Status.DataOBC) == 0 {
-		obc, err := r.Claimer.GetClaim(ctx, edgeDevice.Name, edgeDevice.Namespace)
-		if err != nil {
-			if errors.IsNotFound(err) {
-				logger.Info("Failed to find an existing OBC for the device. Creating new OBC", "edgeDevice", edgeDevice)
-				obc, err = r.Claimer.CreateClaim(ctx, edgeDevice)
-				if err != nil {
-					logger.Error(err, "Cannot create object bucket claim for the device", "EdgeDevice Name", edgeDevice.Name, "EdgeDevice Namespace", edgeDevice.Namespace)
-					return ctrl.Result{Requeue: true}, err
-				}
-			} else {
-				logger.Error(err, "Failed to get OBC for the device", "edgeDevice", edgeDevice)
+	if r.ObcAutoCreate {
+		// create object bucket claim for edge-device
+		if edgeDevice.Status.DataOBC == nil || len(*edgeDevice.Status.DataOBC) == 0 {
+			obc, err := r.createOrGetObc(ctx, edgeDevice)
+			if err != nil {
 				return ctrl.Result{Requeue: true}, err
 			}
-		}
 
-		patch := client.MergeFrom(edgeDevice.DeepCopy())
-		edgeDevice.Status.DataOBC = &obc.Name
-		err = r.EdgeDeviceRepository.PatchStatus(ctx, edgeDevice, &patch)
-		if err != nil {
-			return ctrl.Result{Requeue: true, RequeueAfter: time.Second * 5}, err
+			err = r.addObcReference(ctx, edgeDevice, obc.Name)
+			if err != nil {
+				return ctrl.Result{Requeue: true, RequeueAfter: time.Second * 5}, err
+			}
 		}
 	}
 	return ctrl.Result{}, nil
 }
 
+func (r *EdgeDeviceReconciler) createOrGetObc(ctx context.Context, edgeDevice *managementv1alpha1.EdgeDevice) (*obv1.ObjectBucketClaim, error) {
+	obc, err := r.Claimer.GetClaim(ctx, edgeDevice.Name, edgeDevice.Namespace)
+	if err != nil {
```

Comment:
I would do something like 

```
if err == nil {
logger.Error(err, "Failed to get OBC for the device", "edgeDevice", edgeDevice)
			return nil, err
}
logger := log.FromContext(ctx)
if errors.IsNotFound(err) {
```

It's a bit clear what is the normal flow function.